js/widgets/forms/flipswitch.js

```diff
@@ -251,9 +245,8 @@ return $.widget( "mobile.flipswitch", $.extend( {
 			var currentTheme = options.theme ? options.theme : "inherit",
 				newTheme = options.theme ? options.theme : "inherit";
 
-			this.widget()
-				.removeClass( "ui-bar-" + currentTheme )
-				.addClass( "ui-bar-" + newTheme );
+			this._removeClass( this.flipswitch, null,  "ui-bar-" + currentTheme );
+			this._addClass( this.flipswitch, null,  "ui-bar-" + newTheme );
```

Comment:
Sure enough just tried and this is broken changing the theme option does not remove the old class just adds the new one.